### DISCUSSION TITLE
Is it possible to do manual & multiples Url 301 redirection with vercel?

### DISCUSSION THREAD

**Discussion by MarvingMoreton** [vercel/community#1273] [Help] [answered]
### Summary

Hello, 

I have just migrated my website with multilang and review my site architecture. Hence, I need to do manualy redirection for some URLs. Is there a way to do it with vercel?

To illustrate it :
I want xyz.com/abc to be redirected to xyz.com/en/abc for instance. then xyz.com/blog-article to xyz.com/blog/blog-article.

Is there a wasy to do such thing right now?

It matters a lot for me SEO wise.

Thanks you inb advanced!

**Comment by leerob** [comment, marked as answer]
Yes! This should be possible, either using the native redirects in your framework of choice, or using `vercel.json` otherwise.

**Reply by MarvingMoreton** [reply]
Hello @leerob thanks for the quick reply!

Indeed, I have not done my homework correctly, it is specified in the documentation: https://vercel.com/guides/does-vercel-support-permanent-redirects

Apologies!

However, I have implemented the next.config.js way, and it is not working so far. I've found informal information saying that I should be able to redirect absolute path instead of dynamic. Isn't it correct?

For instance here's my next.config.js:

`
const prismic = require('@prismicio/client');

const sm = require('./sm.json');

const localeOverrides = {
    'fr-wo': 'fr'
};

/** @type {import('next').NextConfig} */
const nextConfig = async () => {
    const client = prismic.createClient(sm.apiEndpoint);

    const repository = await client.getRepository();
    const locales = repository.languages.map(
        (lang) => localeOverrides[lang.id] || lang.id
    );

    return {
        reactStrictMode: true,
        swcMinify: true,

        i18n: {
            // These are all the locales you want to support in
            // your application
            locales,
            // This is the default locale you want to be used when visiting
            // a non-locale prefixed path e.g. `/hello`
            defaultLocale: locales[0],
            // defaultLocale: 'en-ca'
            localeDetection: false
        },
        images: {
            domains: [
                'marving-moreton.cdn.prismic.io',
                'images.prismic.io',
                'res.cloudinary.com'
            ]
        },

        async redirects() {
            return [
                {
                    source: '/a-propos/',
                    destination: '/fr/a-propos',
                    permanent: true
                },
                {
                    source: '/audit-seo/',
                    destination: '/fr/blog/audit-seo',
                    permanent: true
                },
                {
                    source: '/consultant-seo-angers/',
                    destination: '/fr/seo/consultant-seo-angers',
                    permanent: true
                },
                {
                    source: '/consultant-seo-bordeaux/',
                    destination: '/fr/seo/consultant-seo-bordeaux',
                    permanent: true
                },
                {
                    source: '/consultant-seo-freelance/',
                    destination: '/fr/seo/consultant-seo-freelance',
                    permanent: true
                },
                {
                    source: '/consultant-seo-montreal/',
                    destination: '/seo/seo-consultant-montreal',
                    permanent: true
                },
                {
                    source: '/consultant-seo-paris/',
                    destination: '/fr/seo/consultant-seo-paris',
                    permanent: true
                },
                {
                    source: '/consultant-seo-toulouse/',
                    destination: '/fr/seo/consultant-seo-toulouse',
                    permanent: true
                },
                {
                    source: '/consultant-seo/',
                    destination: '/fr/seo',
                    permanent: true
                },
                {
                    source: '/portfolio/flowski-location-ski-canada/',
                    destination: '/fr/seo',
                    permanent: true
                },
                {
                    source: '/portfolio/glace/',
                    destination: '/fr/seo',
                    permanent: true
                },
                {
                    source: '/portfolio/simplemets-blog-culinaire/',
                    destination: 'fr/seo',
                    permanent: true
                },
                {
                    source: '/portfolio/tourcrib-plateforme-reservation/',
                    destination: '/',
                    permanent: true
                },
                {
                    source: '/referencement-naturel-google-gratuit/',
                    destination:
                        '/fr/blog/referencement-naturel-google-gratuit',
                    permanent: true
                },
                {
                    source: '/tendances-seo-en-2021/',
                    destination: '/fr/blog/tendances-seo',
                    permanent: true
                }
            ];
        }
    };
};

module.exports = nextConfig;

`

Thanks in advanced!

**Reply by leerob** [reply]
If you're wanting something more custom than the built-in redirects/i18n, you might consider checking out Middleware! https://nextjs.org/docs/advanced-features/middleware

**Reply by MarvingMoreton** [reply]
All good, it has been fixed!

It was just my last slashes that were preventing the redirection to work… Thanks!
